From ebf175fc7740da0ed97e78ccfee91698d35e5a9a Mon Sep 17 00:00:00 2001
From: Bruno Martins <bgcngm@gmail.com>
Date: Tue, 12 Dec 2017 14:24:27 +0000
Subject: [PATCH 03/14] sepolicy: Include legacy rild policies

 * Disabled in commit c7def121c0a13733060290f9ce1401ca7cf44405,
   but still needed for legacy devices.

 * Addresses the following denials:

     W rild : type=1400 audit(0.0:82): avc: denied { call } for scontext=u:r:rild:s0 tcontext=u:r:servicemanager:s0 tclass=binder permissive=0
     ...
     I Telecom : Event: RecordEntry TC@1: REQUEST_DISCONNECT, null: ICA.dC(InCall package: com.android.dialer)@AHI
     I Telecom : Call: Send disconnect to connection service for call: [TC@1, DIALING, com.android.phone/com.android.services.telephony.TelephonyConnectionService, tel:**********, A, childs(0), has_parent(false), [Capabilities: CAPABILITY_SUPPORT_HOLD CAPABILITY_MUTE CAPABILITY_CANNOT_DOWNGRADE_VIDEO_TO_AUDIO], [Properties:]]: ICA.dC(InCall package: com.android.dialer)@AHI
     W servicemanager: type=1400 audit(0.0:67): avc: denied { search } for name="383" dev="proc" ino=10482 scontext=u:r:servicemanager:s0 tcontext=u:r:rild:s0 tclass=dir permissive=0
     E ServiceManager: SELinux: getpidcon(pid=383) failed to retrieve pid context.
     I ServiceManager: Waiting for service media.audio_flinger...

Change-Id: I83983521162aa9bb63450930d4d242a4f8c5e4c1
---
 legacy-common/rild.te | 4 ++++
 1 file changed, 4 insertions(+)
 create mode 100644 legacy-common/rild.te

diff --git a/legacy-common/rild.te b/legacy-common/rild.te
new file mode 100644
index 0000000..d7c2cde
--- /dev/null
+++ b/legacy-common/rild.te
@@ -0,0 +1,4 @@
+not_full_treble(`
+    binder_use(rild)
+    allow rild { mediaserver_service audioserver_service }:service_manager find;
+')
\ No newline at end of file
-- 
2.7.4

